heise.de.txt

```diff
@@ -8,6 +8,14 @@ author: //span[@class='author']
 
 body: //article | //div[@class='meldung_wrapper'] | //section[@id='artikel_text']
 
+requires_login: yes
+
+login_uri: https://www.heise.de/sso/login/login
+login_username_field: username
+login_password_field: password
+
+not_logged_in_xpath: //body[@Class="a-login__link a-login__link--sso"]
```

Comment:
I'm not sure about that.
It should be a xpath to detect if user needs to log in from the article source. I can't find an element with that xpath in an article page.
I think it might be better to use `//a[@id='heiselogin-link-login']`

Reply:
Ok, my understanding was wrong about the require_login: yes. I thought the login will always be happen and the not_logged_in_xpath must be from the login page [1]. I couldn't even find the Class I have used on the current page, but maybe they have changed it.

I also talked about articles from https://www.heise.de/select, e.g. [2]. Here I can find class="purchase__option-header-container" when you have to login.

[1] https://www.heise.de/sso/login/
[2] https://www.heise.de/select/ix/2019/3/1551687647597726

Reply:
Looks like there is many different places where the login is required.
From that page, it seems that we have to look for `//a[@class="curtain__login-link"]`: https://www.heise.de/tests/Schwarzweiss-Multifunktionsdrucker-fuer-den-Schreibtisch-4294269.html